cf-agent/verify_files.c

```diff
@@ -446,18 +446,6 @@ static PromiseResult VerifyFilePromise(EvalContext *ctx, char *path, const Promi
     }
     else
     {
-        if (!S_ISDIR(osb.st_mode))
-        {
-            if (a.havedepthsearch)
-            {
-                /* TODO: PROMISE_RESULT_DENIED */
-                Log(LOG_LEVEL_DEBUG,
-                    "depth_search (recursion) is promised for a base object '%s' that is not a directory",
-                    path);
```

Comment:
Perhaps a warning, or a verbose log that recursion was promised on a single file, ignoring recursion?

Reply:
Updated.